File: clients/ios/AwareChat-iOS/AwareChat-iOS/Core/Networking/NetworkConfiguration.swift
import Foundation

nonisolated struct NetworkConfiguration: Equatable, Sendable {
  let httpBaseURL: URL
  let webSocketURL: URL

  init(httpBaseURL: URL, webSocketURL: URL) {
    self.httpBaseURL = httpBaseURL
    self.webSocketURL = webSocketURL
  }

  static let localDevelopment = NetworkConfiguration(
    httpBaseURL: URL(string: "http://127.0.0.1:8000")!,
    webSocketURL: URL(string: "ws://127.0.0.1:8000/ws")!
  )
}

File: clients/android/AwareChat-Android/app/src/main/java/com/example/awarechat_android/core/network/NetworkConfiguration.kt
package com.example.awarechat_android.core.network

import okhttp3.HttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrl

data class NetworkConfiguration(
    val httpBaseUrl: HttpUrl,
    val webSocketUrl: String,
) {
    companion object {
        val localDevelopment = NetworkConfiguration(
            httpBaseUrl = "http://10.0.2.2:8000".toHttpUrl(),
            webSocketUrl = "ws://10.0.2.2:8000/ws",
        )
    }
}
